router/http/src/main/scala/io/buoyant/router/http/PathIdentifier.scala

```diff
@@ -20,13 +20,23 @@ case class PathIdentifier(
   def apply(req: Request): Future[RequestIdentification[Request]] =
     req.path.split("/").drop(1) match {
       case path if path.size >= segments =>
+        val params = req.getParams()
```

Comment:
rather than getting and re-encoding the query string, I would just do something like:

```
val query = {
  val u = req.uri
  u.indexOf('?') match {
    case -1 => ""
    case n => u.substring(n)
  }
}
```